lib/bundler/friendly_errors.rb

```diff
@@ -21,6 +21,10 @@ def self.with_friendly_errors
   rescue Thor::Error => e
     Bundler.ui.error e.message
     exit 1
+  rescue NameError => e
```

Comment:
Hm true. Then we could raise a custom exception if the `#require` fails and rescue it here. This way we'll have more information like the gem name, enabling to show more helpful messages.